### ISSUE TITLE
Default Istio.UrlServiceVersion configuration value improvement

### ISSUE THREAD

**Issue by mjobst-necls** [kiali/kiali#4985]
### Problem
Currently, the default value of `UrlServiceVersion` is `http://istiod:15014/version`

https://github.com/kiali/kiali/blob/9bad8de83d2ff207e10c6df1720013a8d5ccf909/config/config.go#L589

This works fine when Kiali runs in the same namespace as Istod.
If this is not the case, `UrlServiceVersion` needs to be set manually

### Suggestion
Set the `UrlServiceVersion` from `http://istiod:15014/version` to `http://istiod.istio-system:15014/version` to also follow the schema of the other URL settings.

**Comment by nrfox** [comment]
That seems like a better default. The change would affect users who have deployed both istiod and kiali in the same namespace other than `istio-system` but I think it's more likely to run kiali outside of the istio-system namespace and the defaults should work for that use case especially given the rest of the urls include the `istio-system` namespace.

@jmazzitelli does the operator need to be updated? Looks like it doesn't specify a value and will just use the server default: https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/defaults/main.yml#L179

**Comment by jmazzitelli** [comment]
I need to think about this. This is changing default behavior, which is not backwards-compatible.

It is usually the case Kiali is deployed in the same namespace as istiod, which may or may not be istio-system. This change would now require Istio installed in istio-system for this default to work.

**Comment by jmazzitelli** [comment]
> This works fine when Kiali runs in the same namespace as Istod.
> If this is not the case, `UrlServiceVersion` needs to be set manually

This is expected behavior. All this request is asking for is a change in documented behavior. Need to consider if this backwards-incompatible change is warranted.

**Comment by jmazzitelli** [comment]
> Looks like it doesn't specify a value and will just use the server default

no, it sets up a default at runtime during reconciliation: https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/tasks/main.yml#L371-L375

And now that I look at that code, changing the server-side default is probably irrelevant under most situations since the operator will explicitly define this value correctly. I think it will be a backwards incompatible change for those using the server helm chart (since the server helm chart doesn't do this same thing that the operator does).

**Comment by jmazzitelli** [comment]
Looking at the operator code, and comparing it with the helm chart, I think my suggestion solution to this enhancement request is to change the server helm chart to mimic what the operator does - set the `url_service_version` value using the `istio_namespace` setting if it is not already set.

**Comment by jmazzitelli** [comment]
You will want to update the server helm chart in a similar way as this: https://github.com/kiali/helm-charts/blob/v1.49.0/kiali-server/templates/_helpers.tpl#L164-L173

(there should be something similar with the url_service_version setting - which then gets set in the configmap: https://github.com/kiali/helm-charts/blob/v1.49.0/kiali-server/templates/configmap.yaml#L27)

Remember - most of those defaults you see in config/config.go are never used (except for unit testing). This is because the operator will always provide a ConfigMap that sets every setting (or virtually every one - there may be one or two that aren't set, I can't remember).

**Comment by jmazzitelli** [comment]
> @jmazzitelli yes this is a backwards incompatible change and users who have deployed kiali and istiod in the same namespace other than `istio-system` would need to update their settings but since the other services all have the format `<service>.istio-system` I don't think the defaults would work for that use case anyways:
> 
> https://github.com/kiali/kiali/blob/9bad8de83d2ff207e10c6df1720013a8d5ccf909/config/config.go#L607
> 
> .

Again, the operator handles this - for prometheus URL for example, see: https://github.com/kiali/kiali-operator/blob/v1.49.0/roles/default/kiali-deploy/tasks/main.yml#L377-L381

If this situation is not handled by the operator (kiali and istiod in the same namespace but not istio-system) then we have a bug we need to fix :) That should already work fine (and I believe it does).
